src/hotspot/cpu/ppc/assembler_ppc.inline.hpp

```diff
@@ -130,6 +143,25 @@ inline void Assembler::divw_(  Register d, Register a, Register b) { emit_int32(
 inline void Assembler::divwu(  Register d, Register a, Register b) { emit_int32(DIVWU_OPCODE  | rt(d) | ra(a) | rb(b) | oe(0) | rc(0)); }
 inline void Assembler::divwu_( Register d, Register a, Register b) { emit_int32(DIVWU_OPCODE  | rt(d) | ra(a) | rb(b) | oe(0) | rc(1)); }
 
+// Prefixed instructions, introduced by POWER10
+inline void Assembler::paddi(  Register d, Register a, long si34, bool r = false) {
+  assert(a != R0 || r, "r0 not allowed, unless R is set (CIA relative)");
+  paddi_r0ok( d, a, si34, r);
+}
+
+inline void Assembler::paddi_r0ok(  Register d, Register a, long si34, bool r = false) {
+  emit_prefix(PADDI_PREFIX_OPCODE | r_eo(r) | d0_eo(si34));
+  emit_int32( PADDI_SUFFIX_OPCODE | rt(d)   | ra(a)   | d1_eo(si34));
+}
+
+inline void Assembler::paddi_or_addi(Register d, Register a, long si34) {
```

Comment:
defined but not used.

Reply:
It is intended to show an example of an efficient set of macros/functions for a prefixed instruction. Since a non-prefixed (32-bit) instruction is more efficient if the argument fits within the operand fields because of smaller I-cache usage and no need for the padding nop, it is convenient to have a macro/function that select better format. I intended to show an example of recommended set of useful macros/functions when adding support for a prefixed instruction. I'm planning to use this function in my next patch. Is it better to move this to the next patch?

Reply:
It would be my preference to post code up only when it's actually used.  When you get around to using it, there's a possibility you will want to modify it a bit, add a comment, etc.   So that can be done all in one commit, instead of split across two.